File: Data Structures & Algorithms/longest-common-prefix/submission-0.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        string s = "";
        int n = 300;
        for(int i = 0; i < strs.size(); i++) {
            int sz = strs[i].size();
            n = min(n, sz);
        }
        for(int i= 0; i < n; i++) {
            
            bool ok = true;
            for(int j = 0; j < strs.size(); j++) {
               
                if(j && strs[j][i] != strs[0][i]) {
                    ok = false;
                    break;
                    
                } 
            }
            if(!ok) break;
            s+= strs[0][i];
        }
        return s;
       
    }
};

File: Data Structures & Algorithms/anagram-groups/submission-0.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        
        groups = {}
        for s in strs:

            freq = [0]*26
            for ch in s:
                freq[ord(ch)-ord('a')]+=1
            
            key = tuple(freq)
            if key not in groups:
                groups[key] = []
            groups[key].append(s)
        return list(groups.values())